internal/controller/ols_app_server_assets.go

```diff
@@ -0,0 +1,262 @@
+		}
+
+		modelConfigs := []ModelConfig{}
+		for _, model := range provider.Models {
+			modelConfig := ModelConfig{
+				Name: model.Name,
+				URL:  model.URL,
+			}
+			modelConfigs = append(modelConfigs, modelConfig)
+		}
+
+		providerConfig := ProviderConfig{
+			Name:            provider.Name,
+			URL:             provider.URL,
+			CredentialsPath: credentialPath,
+			Models:          modelConfigs,
+		}
+		providerConfigs = append(providerConfigs, providerConfig)
+	}
+
+	conversationCache := ConversationCacheConfig{
+		Type:   string(cr.Spec.OLSConfig.ConversationCache.Type),
+		Memory: MemoryCacheConfig{MaxEntries: cr.Spec.OLSConfig.ConversationCache.Memory.MaxEntries},
+	}
+
+	olsConfig := OLSConfig{
+		DefaultModel:    cr.Spec.OLSConfig.DefaultModel,
+		DefaultProvider: cr.Spec.OLSConfig.DefaultProvider,
+		Logging: LoggingConfig{
+			AppLogLevel: cr.Spec.OLSConfig.LogLevel,
+			LibLogLevel: cr.Spec.OLSConfig.LogLevel,
+		},
+		ConversationCache: conversationCache,
+	}
+
+	appSrvConfigFile := AppSrvConfigFile{
+		LLMProviders: providerConfigs,
+		OLSConfig:    olsConfig,
+	}
+	configFileBytes, err := yaml.Marshal(appSrvConfigFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate OLS config file %w", err)
+	}
+
+	configFileHash, err := hashBytes(configFileBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate OLS config file hash %w", err)
+	}
+
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OLSConfigCmName,
+			Namespace: OLSNamespaceDefault,
+			Labels: map[string]string{
+				"app.kubernetes.io/component":  "application-server",
+				"app.kubernetes.io/managed-by": "lightspeed-operator",
+				"app.kubernetes.io/name":       "lightspeed-service-api",
+				"app.kubernetes.io/part-of":    "openshift-lightspeed",
+			},
+			Annotations: map[string]string{
+				OLSConfigHashKey: configFileHash,
+			},
+		},
+		Data: map[string]string{
+			OLSConfigFilename: string(configFileBytes),
+		},
+	}
+
+	return &cm, nil
+}
+
+func (r *OLSConfigReconciler) generateOLSDeployment(cr *olsv1alpha1.OLSConfig) (*appsv1.Deployment, error) {
+
+	// mount points of API key secret
+	const OLSConfigMountPath = "/etc/ols"
+	const OLSConfigVolumeName = "cm-olsconfig"
+	DeploymentSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "application-server",
+		"app.kubernetes.io/managed-by": "lightspeed-operator",
+		"app.kubernetes.io/name":       "lightspeed-service-api",
+		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+	}
+
+	// map from secret name to secret mount path
+	secretMounts := map[string]string{}
+	for _, provider := range cr.Spec.LLMConfig.Providers {
+		credentialMountPath := path.Join(APIKeyMountRoot, provider.CredentialsSecretRef.Name)
+		if strings.ToLower(provider.Name) == "openai" {
+			credentialMountPath = path.Join(APIKeyMountRoot, provider.CredentialsSecretRef.Name)
```

Comment:
this seems to be doing the same thing as the line before the `if` statement?

Reply:
yes, will remove the duplication.